src/javascript/expression.rs

```diff
@@ -113,10 +115,66 @@ impl Generator {
     }
 }
 
+fn int(value: &str) -> Document<'_> {
+    value.to_doc()
```

Comment:
Let's fail if this is a binary, octal, or hex number. Also let's use the binary operator to hint that it's an int

```rust
let x = 1
```
```js
var x = 1 | 0;
```

Reply:
The  binary, octal, or hex all work in JavaScript.
```
0o12 === 10
true
0xF === 15
true
0b11 === 3
true
```

I would argue against providing the integer hint. I was following the reasoning that we should as much as possible generate the JS someone would write. It's not idiomatic handwritten JS to have those hints. Also I thought those hints where useful while Web Assembly was being developed, but if optimising for web assembly wouldn't we just compile to web assembly.

Reply:
Wow that's cool! I have no idea you could do that in JS.

It's not super common, but it will avoid any bugs coming from not having real ints, and it will generate faster code than leaving it off. Microsoft even recommend adding it to hand-written JS https://docs.microsoft.com/en-us/previous-versions/windows/apps/hh781219(v=win.10)?redirectedfrom=MSDN#use-integer-arithmetic-where-possible

Reply:
Added `| 0` for division it also handles the infinity so no explicit check needed.